packages/model-viewer/src/three-components/ModelScene.ts

```diff
@@ -679,6 +679,32 @@ export class ModelScene extends Scene {
     return {position: hit.point, normal: hit.face.normal};
   }
 
+  /**
+   * This method returns the x and y texture-coordinate of the point
+   * on the mesh corresponding to the input pixel coordinates given relative to
+   * the model-viewer element. If the mesh is not hit, the result is null.
+   */
+  uvFromPoint(ndcPosition: Vector2, object: Object3D = this): Vector2|null {
```

Comment:
Since ModelScene is an internal class, we're free to refactor it. Let's combine this method with the one above and return a struct with everything to avoid some duplication.